File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionFormParse.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;
import co.nlighten.jsontransform.formats.formurlencoded.FormUrlEncodedFormat;

public class TransformerFunctionFormParse extends TransformerFunction {

    public TransformerFunctionFormParse() {
        super();
    }

    @Override
    public Object apply(FunctionContext context) {
        // TODO: how to create the format once?
        return new FormUrlEncodedFormat(context.getAdapter()).deserialize(context.getString(null));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionFormParse.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import FormUrlEncodedFormat from "../formats/formurlencoded/FormUrlEncodedFormat";

class TransformerFunctionFormParse extends TransformerFunction {
  private static readonly FORM_URL_ENCODED_FORMAT = new FormUrlEncodedFormat();

  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const value = await context.getString(null);
    if (value === null) return null;
    return TransformerFunctionFormParse.FORM_URL_ENCODED_FORMAT.deserialize(value);
  }
}

export default TransformerFunctionFormParse;
